src/plugins/intel_cpu/src/nodes/strided_slice.cpp

```diff
@@ -312,51 +285,87 @@ bool StridedSlice::isExecutable() const {
 }
 
 void StridedSlice::createPrimitive() {
-    if (!isExecutable()) {
-        return;
+    if (inputShapesDefined() && isExecutable() && !shapeHasDataDependency) {
+        if (needPrepareParams()) {
+            prepareParams();
+        }
+        updateLastInputDims();
     }
-    auto& dstMemPtr = getChildEdgeAt(0)->getMemoryPtr();
-    auto& srcMemPtr = getParentEdgeAt(DATA_ID)->getMemoryPtr();
-    if (!dstMemPtr || !dstMemPtr->isAllocated())
-        THROW_ERROR << "has not allocated destination memory.";
-    if (!srcMemPtr || !srcMemPtr->isAllocated())
-        THROW_ERROR << "has not allocated input memory.";
-    if (getSelectedPrimitiveDescriptor() == nullptr)
-        THROW_ERROR << "has unidentified preferable primitive descriptor.";
+}
 
-    if (!srcMemPtr->getDesc().hasLayoutType(LayoutType::ncsp))
-        orderParametersByLayouts(srcMemPtr);
+bool StridedSlice::needPrepareParams() const {
+    return true;
+}
 
-    if (inputShapesDefined()) {
-        prepareParams();
-        updateLastInputDims();
+void StridedSlice::prepareParams() {
+    updateLastInputDims();
+
+    std::vector<MemoryCPtr> srcMemory;
+    for (int i = 0; i < getOriginalInputsNumber(); i++) {
+        srcMemory.push_back(getParentEdgeAt(i)->getMemoryPtr());
     }
+    execPtr = std::make_shared<StridedSliceCommonExecutor>(attrs, srcMemory, getChildEdgeAt(0)->getMemoryPtr(), errorPrefix);
 }
 
-void StridedSlice::orderParametersByLayouts(const MemoryPtr& srcMemPtr) {
+bool StridedSlice::needShapeInfer() const {
+    return Node::inputShapesModified() || shapeHasDataDependency;
+}
+
+void StridedSlice::execute(dnnl::stream strm) {
+    if (!execPtr)
+        IE_THROW() << errorPrefix << "doesn't have compiled executor!";
+
+    std::vector<MemoryCPtr> srcMemory;
+    for (int i = 0; i < getOriginalInputsNumber(); i++) {
+        srcMemory.push_back(getParentEdgeAt(i)->getMemoryPtr());
+    }
+
+    execPtr->exec(srcMemory, getChildEdgeAt(0)->getMemoryPtr());
```

Comment:
Good point. Implemented in bca550936e84d139c5a01c70a866ba0de51fa407.

Regarding raw pointers. The impl is in general aligned with current view in Executor interface: executor might be created one time with undefinned shapes and called many times with target shapes. We cannot do it for StridedSlice due to dependency on input data, so recreate it each time.
1. Ctor takes parameters reuqired to create the executor and memory objects to have access to blocked memory desc. We cannot limit the API to memory descriptors only because need access to Begin, End and Strides memory. This is discussable moment - need rethinking in future.
2. Exec takes memory object since need to have access to defined target shapes (and as a consicuence defined memory descs) besides the data itself.